src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -773,22 +770,22 @@ bool ClusterTaskManager::AnyPendingTasks(Task *exemplar, bool *any_pending,
   // We are guaranteed that these tasks are blocked waiting for resources after a
   // call to ScheduleAndDispatchTasks(). They may be waiting for workers as well, but
   // this should be a transient condition only.
-  for (const auto &shapes_it : boost::join(tasks_to_dispatch_, tasks_to_schedule_)) {
-    auto &work_queue = shapes_it.second;
-    for (const auto &work_it : work_queue) {
-      const auto &task = std::get<0>(work_it);
-      if (task.GetTaskSpecification().IsActorCreationTask()) {
-        *num_pending_actor_creation += 1;
-      } else {
-        *num_pending_tasks += 1;
-      }
+  std::function<void(const Work &)> fn = [exemplar, any_pending,
```

Comment:
Usually how often do we call this function? I feel it might be very expensive in some cases since there is no early return, which means that if there are a lot of tasks, it's going to take a long time to finish.

Reply:
Changed it to `std::function<bool(const Work &)>` to allow short circuiting.